yt/chyt/server/query_context-inl.h

```diff
@@ -0,0 +1,28 @@
+#ifndef QUERY_CONTEXT_INL_H_
+#error "Direct inclusion of this file is not allowed, include format.h"
+// For the sake of sane code completion.
+#include "query_context.h"
+#endif
+
+namespace NYT::NClickHouseServer {
+
+////////////////////////////////////////////////////////////////////////////////
+
+template <class T>
+void TQueryContext::SetRuntimeVariable(const TString& key, const T& value)
+{
+    auto guard = Guard(QueryLogLock_);
+    YT_VERIFY(RuntimeVariables_);
+    if (RuntimeVariables_->Contains(key)) {
+        YT_LOG_WARNING(
```

Comment:
Just in case:
TQueryContext is shared between all subqueries in the query.
E.g. if you have query like this:
```sql
SELECT * FROM t1
ORDER BY key
LIMIT 10
UNION ALL
SELECT * FROM t2
ORDER BY key
LIMIT 10
```
You will override the runtime variables when executing the second subquery and trigger this `YT_LOG_WARNING`

Reply:
After our discussion, I don't think it is worth investing into fixing this. Changed to YT_LOG_DEBUG, so that we could see it in the log.